<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerInteractiveTubesToTreeModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerInteractiveTubesToTreeModuleWidget">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>345</width>
    <height>325</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="ctkCollapsibleButton" name="IOCollapsibleButton" native="true">
     <property name="text" stdset="0">
      <string>IO</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="horizontalSpacing">
       <number>9</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="InputLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Input TRE file containing disjoint tubes</string>
        </property>
        <property name="text">
         <string>Input TRE File</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="InputSpacialObjectsNodeComboBox" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Input TRE file containing disjoint tubes</string>
        </property>
        <property name="nodeTypes" stdset="0">
         <stringlist>
          <string>vtkMRMLSpatialObjectsNode</string>
         </stringlist>
        </property>
        <property name="baseName" stdset="0">
         <string/>
        </property>
        <property name="addEnabled" stdset="0">
         <bool>false</bool>
        </property>
        <property name="editEnabled" stdset="0">
         <bool>false</bool>
        </property>
        <property name="renameEnabled" stdset="0">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="OutputLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Output TRE file containing a tree of tubes</string>
        </property>
        <property name="text">
         <string>Output TRE File </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLNodeComboBox" name="OutputSpacialObjectsNodeComboBox" native="true">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Output TRE file containing a tree of tubes</string>
        </property>
        <property name="nodeTypes" stdset="0">
         <stringlist>
          <string>vtkMRMLSpatialObjectsNode</string>
         </stringlist>
        </property>
        <property name="showHidden" stdset="0">
         <bool>true</bool>
        </property>
        <property name="baseName" stdset="0">
         <string>Output TRE File</string>
        </property>
        <property name="renameEnabled" stdset="0">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="AlgorithmParametersCollapsibleButton" native="true">
     <property name="text" stdset="0">
      <string>Algorithm Parameters</string>
     </property>
     <property name="collapsed" stdset="0">
      <bool>false</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_1">
      <item row="0" column="0">
       <widget class="QLabel" name="MaxTubeDistanceToRadiusRatioLabel">
        <property name="text">
         <string>Maximum Tube Distance To Radius Ratio</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="0" column="2">
       <widget class="qMRMLSliderWidget" name="MaxTubeDistanceToRadiusSliderWidget" native="true">
        <property name="toolTip">
         <string>Maximum allowed distance to radius ratio to connect tubes</string>
        </property>
        <property name="minimum" stdset="0">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum" stdset="0">
         <double>200.000000000000000</double>
        </property>
        <property name="value" stdset="0">
         <double>2.000000000000000</double>
        </property>
        <property name="tickInterval" stdset="0">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="MaxContinuityAngleErrorLabel">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Maximum Continuity Angle Error</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="2">
       <widget class="qMRMLSliderWidget" name="MaxContinuityAngleErrorSliderWidget" native="true">
        <property name="toolTip">
         <string>Maximum continuity angle error</string>
        </property>
        <property name="minimum" stdset="0">
         <double>100.000000000000000</double>
        </property>
        <property name="maximum" stdset="0">
         <double>300.000000000000000</double>
        </property>
        <property name="value" stdset="0">
         <double>180.000000000000000</double>
        </property>
        <property name="tickInterval" stdset="0">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="RemoveOrphanTubesLabel">
        <property name="text">
         <string>Remove Orphan Tubes</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>30</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="2">
       <widget class="QCheckBox" name="RemoveOrphanTubesCheckBox">
        <property name="toolTip">
         <string>Specify if orphan tubes should be removed</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="RootTubeIdListLabel">
        <property name="text">
         <string>Root Tube ID List</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLineEdit" name="RootTubeIDListLineEdit">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>List of root tube Ids</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="4" column="2">
       <widget class="QPushButton" name="PickRootTubesPushButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="toolTip">
         <string>Place the fiducial on the root tube</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="PickRootTubesLabel">
        <property name="text">
         <string>Pick Root Tubes</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <spacer name="horizontalSpacer_6">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
     <widget class="qSlicerInteractiveTubesToTreeTableWidget" name="Table" native="true"/>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="qSlicerCLIProgressBar" name="CLIProgressBar" native="true"/>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="RestoreDefaultsPushButton">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Restore Defaults</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="CancelPushButton">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ApplyPushButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Apply</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSliderWidget</class>
   <extends>ctkSliderWidget</extends>
   <header>qMRMLSliderWidget.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerCLIProgressBar</class>
   <extends>QWidget</extends>
   <header>qSlicerCLIProgressBar.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>qSlicerInteractiveTubesToTreeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>InputSpacialObjectsNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>222</x>
     <y>280</y>
    </hint>
    <hint type="destinationlabel">
     <x>195</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerInteractiveTubesToTreeModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>OutputSpacialObjectsNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>273</y>
    </hint>
    <hint type="destinationlabel">
     <x>408</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>currentNodeChanged(vtkMRMLNode*)</signal>
  <slot>setSpatialObjectsNode(vtkMRMLNode*)</slot>
 </slots>
</ui>
